inlang/source-code/sdk/src/createMessagesQuery.ts

```diff
@@ -339,6 +344,7 @@ async function loadMessagesViaPlugin(
 				messages.set(loadedMessageClone.id, loadedMessageClone)
 				// NOTE could use hash instead of the whole object JSON to save memory...
 				messageState.messageLoadHash[loadedMessageClone.id] = importedEnecoded
+				loadedMessageCount++
```

Comment:
You're bumping the count in 2 code paths but only testing for `loadedMessageCount % messagesPerTick === 0`  in one of those paths. This makes it hard to know if the check will actually be hit.

Why not test for > maxMessagesPerTick and then reset the count?